src/utils/commondir.ts

```diff
@@ -0,0 +1,21 @@
+// ported from https://github.com/substack/node-commondir
+export default function commondir (relfiles: string[]) {
+    var files = relfiles;
+
+    var res = files.slice(1).reduce(function (ps, file) {
+        if (!file.match(/^([A-Za-z]:)?\/|\\/)) {
+            throw new Error('relative path without a basedir');
+        }
```

Comment:
If we know that our inputs will pass this, it may be possible to remove this check.

Reply:
removed